policyeval/base_worker.go

```diff
@@ -120,13 +115,7 @@ func (w *BaseWorker) handlePolicy(ctx context.Context, eval *sdk.ScalingEvaluati
 	// Fetch target status.
 	logger.Debug("fetching current count")
 
-	currentStatus, err := w.runTargetStatus(targetInst, eval.Policy)
-	if err != nil {
-		return fmt.Errorf("failed to fetch current count: %v", err)
-	}
-	if !currentStatus.Ready {
-		return errTargetNotReady
-	}
+	currentStatus := eval.TargetStatus
```

Comment:
Hum...I don't quite understand why this was removed. The `eval.TargetStatus` here may contain stale data, since the eval enqueue and dequeue are async operations, so by the time this eval is handled by the worker it may not be ready anymore, so we must check again before attempting any scaling action.

Reply:
Does it makes sense to have it on both places?

Reply:
I would say so. We check on equeue to make sure the policy is still valid and avoid unnecessary work by processing evals for targets that are not ready:
https://github.com/hashicorp/nomad-autoscaler/blob/516536afcecdc688cb6f5be4301dcf79a89c04c5/policy/handler.go#L288-L309

And then recheck on dequeue to make sure the assumptions queue time still hold.

If we were to remove one of them, I would keep this one since it's where scaling actions actually happen, but the policy handler already needs to query the target, so we already have the opportunity to check the status early there, so there's no harm in doing it.

Reply:
I tried to made it obvious, because otherwise it just looks like a duplication :)